pkg/kindsys/EXTENDING.md

```diff
@@ -0,0 +1,60 @@
+# Kind System
```

Comment:
Threw down a quick first pass at a guide to extending the kind system. Something more durable than comments in this PR, and reflects how i expect maintenance of this to work.

This lacks [the concrete example i gave in this comment](https://github.com/grafana/grafana/pull/56492#issuecomment-1284791083), but that can be added later.

probably most relevant for @sakjur @ryantxu @ArturWierzbicki @papagian @mckn @IfSentient   @joanlopez @xlson

Reply:
oh and @wbrowne, this doc starts to cover what we talked about the other day

Reply:
Super! Thanks @sdboyer 🙏🏻

Reply:
ahh and @yuri-tceretian! this is what you'd use to add a thing for that contextual logging capability we discussed in backend arch meeting some months back